### PR TITLE
Feat/token gated reference module

### PR DESCRIPTION
**Pull request by vicnaum**
TokenGatedReferenceModule implementation with minimal tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,38 @@
...
+## Collect modules
+
+- [**Auction Collect Module**](./contracts/collect/AuctionCollectModule.sol): This module works by creating an English auction for the underlying publication. After the auction ends, only the auction winner is allowed to collect the publication.
+- [**Updatable Ownable Fee Collect Module**](./contracts/collect/UpdatableOwnableFeeCollectModule.sol): A fee collect module that, for each publication that uses it, mints an ERC-721 ownership-NFT to its author. Whoever owns the ownership-NFT has the rights to update the parameters required to do a successful collect operation over its underlying publication.
+
+## Follow modules
+
+## Reference modules
+
+- [**Token Gated Reference Module**](./contracts/reference/TokenGatedReferenceModule.sol): A reference module that validates that the user who tries to reference has a required minimum balance of ERC20/ERC721 token.
```
Comment (donosonaumczuk) [inline README.md]:
```suggestion
## Collect modules

- [**Aave Fee Collect Module**](./contracts/collect/AaveFeeCollectModule.sol): Extend the LimitedFeeCollectModule to deposit all received fees into the Aave Polygon Market (if applicable for the asset) and send the resulting aTokens to the beneficiary.
- [**Auction Collect Module**](./contracts/collect/AuctionCollectModule.sol): This module works by creating an English auction for the underlying publication. After the auction ends, only the auction winner is allowed to collect the publication.
- [**Updatable Ownable Fee Collect Module**](./contracts/collect/UpdatableOwnableFeeCollectModule.sol): A fee collect module that, for each publication that uses it, mints an ERC-721 ownership-NFT to its author. Whoever owns the ownership-NFT has the rights to update the parameters required to do a successful collect operation over its underlying publication.

## Follow modules

## Reference modules

- [**Degrees Of Separation Reference Module**](./contracts/reference/DegreesOfSeparationReferenceModule.sol): This reference module allows to set a degree of separation `n`, and then allows to comment/mirror only to profiles that are at most at `n` degrees of separation from the author of the root publication. 
- [**Token Gated Reference Module**](./contracts/reference/TokenGatedReferenceModule.sol): A reference module that validates that the user who tries to reference has a required minimum balance of ERC20/ERC721 token.
```

File: contracts/reference/TokenGatedReferenceModule.sol [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    ) external override onlyHub returns (bytes memory) {
+        GateParams memory gateParams = abi.decode(data, (GateParams));
+
+        // Checking if the tokenAddress resembles ERC20/ERC721 token (by calling balanceOf() function)
+        (bool success, bytes memory result) = gateParams.tokenAddress.staticcall(
+            abi.encodeWithSignature('balanceOf(address)', address(this))
+        );
+        // We don't check if the contract exists cause we expect the return data anyway
+        if (gateParams.minThreshold == 0 || !success || result.length == 0 || result.length > 32)
+            revert Errors.InitParamsInvalid();
```
Comment (donosonaumczuk) [inline contracts/reference/TokenGatedReferenceModule.sol]: This must be tested to make sure I'm not wrong, but I guess we can do:

```solidity
uint256 internal constant UINT256_BYTES = 32;
```
And then:
```suggestion
        if (gateParams.minThreshold == 0 || !success || result.length != UINT256_BYTES)
            revert Errors.InitParamsInvalid();
```

But I'm assuming the answer is always padded to fit the return type (uint256),

**Review by donosonaumczuk — CHANGES_REQUESTED** [review]
Looks really good man, great job!

File: contracts/reference/TokenGatedReferenceModule.sol (hunk 2 above)
Comment (vicnaum) [reply contracts/reference/TokenGatedReferenceModule.sol]: Yes, that's correct - even `returns (uint8)` length is 32 bytes.

**Review by donosonaumczuk — APPROVED** [review]
Looks great

**Pull request merged**